### ISSUE TITLE
Gutenberg: Masonry gallery not rendering correctly upon publication

### ISSUE THREAD

**Issue by andreabadgley19** [Automattic/wp-calypso#36905]
Customer in 2430093-zen reported overlapping and irregularly spaced images in masonry gallery. I was able to reproduce this on a test site in both the user's Dyad 2 theme and the default Twenty Seventeen theme.

#### Steps to reproduce
These were done in Twenty Seventeen theme.
1. Starting at URL: https://wordpress.com/block-editor/post/atomicbadger.blog
2. Add a gallery block and select masonry.
3. Add several images, the more the better to be able to view the overlapping and spacing
4. Gallery looks great inside the block in the editor
5. Click Preview and the gallery looks great in the preview that pops up and when you click Visit Site to preview.
6. Click Update (or Publish), and published gallery shows overlapping and gapped images.
7. Here's a link: https://atomicbadger.blog/2019/10/20/masonry-gallery-gutenberg/

#### What I expected
I expected the gallery to have proper spacing between the images as it shows in the editor.

#### What happened instead
Published gallery shows overlapping images and gaps. See screenshots below.

#### Browser / OS version
Chrome v77.0.3865.120. Also renders incorrectly in Safari in iOS 13.1.3.

#### Screenshot / Video
What it looks like in preview:
![masonry gallery preview](https://user-images.githubusercontent.com/56366469/67161668-af7ea780-f32a-11e9-9795-a7df6406819e.png)

What it looks like on published post:
![masonry gallery](https://user-images.githubusercontent.com/56366469/67161490-0b483100-f329-11e9-9836-90f5f1576547.png)

#### Context / Source
The default gallery appears to be working correctly, so I've advised the user to switch to that instead of the masonry gallery.

**Comment by simison** [comment]
the Tiled gallery columns layout with two columns gives a very similar layout so they could also use that.

Looks like the gallery starts working when resizing it, so there's prolly something running too early.

**Comment by Robertght** [comment]
@andreabadgley19 does this look like this one: https://github.com/Automattic/wp-calypso/issues/36365 ?

**Comment by andreabadgley19** [comment]
@robertght It does look similar. CoBlock is installed on the site I tested in. My image results and the customer's image results look different from what's shown in the screenshots in #36365 , in that there is sometimes smushing and sometimes big gaps as seen in the screenshot above. That could be theme related or device related, though.

This is what I used to create the gallery:

![Masonry gallery](https://user-images.githubusercontent.com/56366469/67518465-dadafc80-f672-11e9-8e9f-36326f98a6f0.png)

This is the result of the gallery I created from that screenshot (in Mayland theme):
![Mayland masonry gallery](https://user-images.githubusercontent.com/56366469/67518544-f940f800-f672-11e9-8ee8-3e07105be986.png)

And this is what it should look like:
![masonry preview](https://user-images.githubusercontent.com/56366469/67518638-2ee5e100-f673-11e9-8871-7af52da8badf.png)
